atomic_reactor/plugins/post_flatpak_create_oci.py

```diff
@@ -75,112 +41,52 @@ def __init__(self, workflow):
-        # to get content of the image
-        # OSBS2 TBD
-        # container_dict = self.tasker.create_container(image, command=["/bin/bash"])
-        # container_id = container_dict['Id']
-
-        # try:
-        #     return self._export_container(container_id)
-        # finally:
-        #     self.log.info("Cleaning up docker container")
-        #     self.tasker.remove_container(container_id)
-        return None, None
-
-    def _get_oci_image_id(self, oci_path):
-        cmd = ['skopeo', 'inspect', '--raw', 'oci:{}'.format(oci_path)]
-        raw_manifest = retries.run_cmd(cmd)
-        oci_image_manifest = json.loads(raw_manifest)
-        return oci_image_manifest['config']['digest']
-
-    def _copy_oci_to_local_storage(self, oci_path, name, tag):
-        """Copy OCI image to internal container storage
-
-        The internal storage to copy the image to is defined by the workflow.
-
-        :param oci_path: str, path to OCI directory
-        :param name: str, name to be given to the image in the internal storage
-        :param tag: str, tag to apply to name in the internal storage
-        """
-        skopeo_copy_dst = '{}:{}:{}'.format(self.workflow.storage_transport, name, tag)
-        cmd = ['skopeo', 'copy', 'oci:{}'.format(oci_path), skopeo_copy_dst]
-
-        self.log.info("Copying built image to internal container image storage as %s:%s", name, tag)
-        try:
-            retries.run_cmd(cmd)
-        except subprocess.CalledProcessError as e:
-            self.log.error("image copy failed with output:\n%s", e.output)
-            raise
-
-    def run(self):
-        if not is_flatpak_build(self.workflow):
-            self.log.info('not flatpak build, skipping plugin')
-            return
-
-        source = get_flatpak_source_info(self.workflow)
-        if source is None:
-            raise RuntimeError("flatpak_create_dockerfile must be run before flatpak_create_oci")
-
-        self.builder = FlatpakBuilder(source, self.workflow.source.workdir,
+    def build_flatpak_image(self, source, build_dir: BuildDir) -> Dict[str, Union[str, int]]:
+        self.builder = FlatpakBuilder(source, build_dir.path,
                                       'var/tmp/flatpak-build',
                                       parse_manifest=parse_rpm_output,
                                       flatpak_metadata=self.flatpak_metadata)
 
-        df_labels = df_parser(self.workflow.df_path, workflow=self.workflow).labels
-        self.builder.add_labels(df_labels)
+        df_labels = build_dir.dockerfile_with_parent_env(
+            self.workflow.imageutil.base_image_inspect()
+        ).labels
 
-        tarred_filesystem, manifest = self._export_filesystem()
-        self.log.info('filesystem tarfile written to %s', tarred_filesystem)
-        self.log.info('manifest written to %s', manifest)
+        self.builder.add_labels(df_labels)
 
-        image_components = self.builder.get_components(manifest)
```

Comment:
was this useful? will this return a result different from `rpmqa` and if so should we update https://github.com/containerbuildsystem/atomic-reactor/blob/master/atomic_reactor/utils/koji.py#L422?

Reply:
Hmm, good question. I'd say we should, since we're relying on the logic of FlatpakBuilder here so we cannot guarantee that the results will *not* be different

Reply:
And OSBS 1 did use the results from flatpak_create_oci - at first I thought that wasn't the case, because `rpmqa` runs later, but it has a workaround just for this
https://github.com/containerbuildsystem/atomic-reactor/blob/d7ec9ef5ba2d230aa70d12905f37952a1d2f1414/atomic_reactor/plugins/post_rpmqa.py#L43-L45

Reply:
updated to use image components from results of this plugin